db/migrate/20141230222755_create_group_members.rb

```diff
@@ -0,0 +1,13 @@
+class CreateGroupMembers < ActiveRecord::Migration
+  def change
+    create_table :group_members do |t|
+      t.references :user, index: true, null: false
+      t.references :group, index: true, null: false
+      t.boolean :admin, null: false, default: false
+      t.boolean :pending, null: false, default: true
+
+      t.index [:user_id, :group_id], unique: true
```

Comment:
But we do lookups individually on both user and group id quite often, why wouldn't we have indexes on the two main lookup keys?